src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+    }
+
+    private async Task<AsyncDisposableValue<string>> PrepareVmrForkAsync(
+        string branch,
+        string productRepoUri,
+        string productRepoForkUri)
+    {
+        // Sync the VMR fork branch
+        await SyncForkAsync("dotnet", "dotnet", branch);
+        // Check if the user has the forked VMR in local DB
+        await AddRepositoryToBarIfMissingAsync(VmrForkUri);
+
+        // Create a temporary branch
+        var newBranchName = Guid.NewGuid().ToString();
+        var baseBranch = await ghClient.Git.Reference.Get(MaestroAuthTestOrgName, VmrForkRepoName, $"heads/{branch}");
+        var newBranch = new NewReference($"refs/heads/{newBranchName}", baseBranch.Object.Sha);
+        await ghClient.Git.Reference.Create(MaestroAuthTestOrgName, VmrForkRepoName, newBranch);
+
+
+        // Fetch source mappings and source manifest files and replace the mapping for the repo we're testing on        
+        await UpdateRemoteFileAsync(newBranchName, productRepoUri, productRepoForkUri, SourceMappingsPath);
+        await UpdateRemoteFileAsync(newBranchName, productRepoUri, productRepoForkUri, SourceManifestPath);
+
+        return AsyncDisposableValue.Create(newBranchName, async () =>
+        {
+            logger.LogInformation("Cleaning up temporary branch {branchName}", newBranchName);
+            try
+            {
+                await DeleteGitHubBranchAsync(VmrForkRepoName, newBranchName);
+            }
+            catch (Exception)
+            {
+                // If this throws an exception the most likely cause is that the branch was already deleted
+            }
+        });
+    }
+
+    private async Task DeleteGitHubBranchAsync(string repo, string branch) => await ghClient.Git.Reference.Delete(MaestroAuthTestOrgName, repo, $"heads/{branch}");
+
+    private async Task UpdateRemoteFileAsync(string branch, string productRepoUri, string productRepoForkUri, string filePath)
+    {
+        // Fetch remote file and replace the product repo URI with the repo we're testing on        
+        var sourceMappingsFile = (await ghClient.Repository.Content.GetAllContentsByRef(
+            MaestroAuthTestOrgName,
+            VmrForkRepoName,
+            filePath,
+            branch)).FirstOrDefault() ??
+                throw new Exception($"Failed to find file {SourceMappingsPath} in {MaestroAuthTestOrgName}" +
+                    $"/{VmrForkRepoName} on branch {SourceMappingsPath}");
+
+        // Replace the product repo uri with the forked one
+        var updatedSourceMappings = sourceMappingsFile.Content.Replace(productRepoUri, productRepoForkUri);
+        UpdateFileRequest update = new(
+            $"Update {productRepoUri} source mapping",
+            updatedSourceMappings,
+            sourceMappingsFile.Sha,
+            branch);
+
+        await ghClient.Repository.Content.UpdateFile(
+            MaestroAuthTestOrgName,
+            VmrForkRepoName,
+            filePath,
+            update);
```

Comment:
What happens if we update a repo branch with custom changes and later the tool tries to sync the fork again? Will it reset?

Reply:
You mean if someone makes changes to the fork, and then we try to sync? They won't reset currently, we'll either merge successfully or have a merge conflict. We should probably try to reset the fork

Reply:
I made the Sync method always reset the branch when syncing